corehq/apps/locations/static/locations/js/location_tree.async.js

```diff
@@ -1,410 +1,455 @@
+                return;
+            }
+            return new LocationModel({
+                uuid: model.selected_location_id(),
+                can_edit: options.can_edit_root,
+                is_archived: options.show_inactive,
+                reloadLocationSearchSelect: options.reloadLocationSearchSelect,
+                clearLocationSelection: options.clearLocationSelection,
+                new_loc_url: options.new_loc_url,
+                loc_edit_url: options.loc_edit_url,
+            }, this);
+        });
 
-    this.set_children = function(data) {
-        var children = [];
-        if (data) {
-            children = _.sortBy(data, function(e) {
-                return e.name;
-            });
-        }
+        this.selected_location_tree = tree_model;
 
-        if (loc.children().length > 0 && loc.name() !== "_root") {
-            for (var child_idx = 0; child_idx < children.length; child_idx++) {
-                if (children[child_idx].name === loc.children()[0].name()) {
-                    children.splice(child_idx, 1);
-                    break;
-                }
+        this.lineage = ko.computed(function() {
+            if (!model.selected_location()) {
+                return;
             }
+            $.ajax({
+                type: 'GET',
+                url: model.selected_location().loc_lineage_url(model.selected_location().uuid()),
+                dataType: 'json',
+                error: 'error',
+                success: function(response) {
+
+                    var expand_tree = function() {
+                        var child = null;
+                        for (var lineage_idx = 0; lineage_idx < response.lineage.length; lineage_idx++) {
+                            var location = response.lineage[lineage_idx];
+                            var data = {
+                                name: location.name,
+                                location_type: location.location_type,
+                                uuid: location.location_id,
+                                is_archived: location.is_archived,
+                                can_edit: options.can_edit_root,
+                                children: child,
+                                expanded: child ? 'semi' : false,
+                                children_status: 'semi_loaded',
+                                reloadLocationSearchSelect: options.reloadLocationSearchSelect,
+                                clearLocationSelection: options.clearLocationSelection,
+                            };
+                            var level = new LocationModel(data, model.selected_location_tree, response.lineage.length - lineage_idx - 1);
+                            child = Array.of(Object.assign({}, data));
+                        }
+                        var root_children = [];
+                        for (var child_idx = 0; child_idx < this.selected_location_tree.locs.length; child_idx++) {
```

Comment:
This code used to reference `locs` (global?!) but now references `this.selected_location_tree.locs` (also below). It's not clear to me what `this` is bound to in this context; looks like it may be `undefined` given how it's called: `model.selected_location_tree.root(expand_tree());` Might be clearer to use `tree_model.locs`, which references the variable [assigned to `this.selected_locations_tree` above on line 106](https://github.com/dimagi/commcare-hq/pull/19654/files#diff-873082aa0e19d3c4d260063223f38141R106)

I found this using [?w=1](https://github.com/dimagi/commcare-hq/pull/19654/files?w=1) to view the diff, but it's kinda hard to match up the lines in the non-whitespace-ignoring normal PR view to make this comment, so maybe I'm mistaken?